### ISSUE TITLE
Automate release build

### ISSUE THREAD

**Issue by frankscholten** [ContainerSolutions/minimesos#113]
- [x] Build jar
- [x] Builds the Docker images
- [x] Pushes Docker images to hub with correct tag
- [x] add image versioning to ```minimesos``` script

**Comment by lguminski** [comment]
The current (manual) process is sensitive to tagging mistakes. ES system tests got affected.

**Comment by frankscholten** [comment]
First automate this process and run it on Jenkins. In 0.9.0 we will configure Travis.

**Comment by sadovnikov** [comment]
Added by `id 'net.researchgate.release' version '2.3.5'` plugin tasks
```
Release tasks
-------------
afterReleaseBuild - Runs immediately after the build when doing a release
beforeReleaseBuild - Runs immediately before the build when doing a release
checkCommitNeeded - Checks to see if there are any added, modified, removed, or un-versioned files.
checkSnapshotDependencies - Checks to see if your project has any SNAPSHOT dependencies.
checkUpdateNeeded - Checks to see if there are any incoming or outgoing changes that haven't been applied locally.
commitNewVersion - Commits the version update to your SCM
confirmReleaseVersion - Prompts user for this release version. Allows for alpha or pre releases.
createReleaseTag - Creates a tag in SCM for the current (un-snapshotted) version.
createScmAdapter - Finds the correct SCM plugin
initScmAdapter - Initializes the SCM plugin
preTagCommit - Commits any changes made by the Release plugin - eg. If the unSnapshotVersion tas was executed
release - Verify project, release, and update version to next.
runBuildTasks - Runs the build process in a separate gradle run.
unSnapshotVersion - Removes "-SNAPSHOT" from your project's current version.
updateVersion - Prompts user for the next version. Does it's best to supply a smart default.
```

**Comment by frankscholten** [comment]
@sadovnikov Once this is finished can remove the `afterEvaluate` block from `minimesos/build.gradle`?

**Comment by sadovnikov** [comment]
Next version of the project is written in `gradle.properties` file. Between releases `bin/minimesos` refers to `latest` image of minimesos

Release will result is two commits:
![image](https://cloud.githubusercontent.com/assets/4749332/13992401/40724c52-f11c-11e5-8479-00787a784eeb.png)

The first commit updates `bin/minimesos`
![image](https://cloud.githubusercontent.com/assets/4749332/13992481/8dca20ba-f11c-11e5-9106-18744147045c.png)

The second one happens after the code is tagged and it reverts update in `bin/minimesos` and sets new version in `gradle.properties`
![image](https://cloud.githubusercontent.com/assets/4749332/13992533/be9ae760-f11c-11e5-88d5-65daeaff41e6.png)

The build also tags and pushes `minimesos` image to Docker Hub
```
Pushing containersol/minimesos:0.9.2 docker image to Docker Hub...
Pushing containersol/minimesos:latest docker image to Docker Hub...
```

There are still a few things to complete

- [ ] trigger generation of new documentation
- [ ] trigger JitPack build

See implementation in PR #333

**Comment by sadovnikov** [comment]
Execution line
```
./gradlew release -PgithubAuthToken=<GITHUB AUTHENTICATION TOKEN>
```

**Comment by sadovnikov** [comment]
resolved in #337. See https://github.com/ContainerSolutions/minimesos/wiki/ReleaseBuild
